packages/server/src/conversations/api.ts

```diff
@@ -1,83 +1,54 @@
-import { Router } from 'express'
-import { Auth } from '../base/auth/auth'
+import { uuid } from '@botpress/messaging-base'
+import { Response } from 'express'
+import { ApiManager } from '../base/api-manager'
+import { ClientApiRequest } from '../base/auth/client'
 import { Schema } from './schema'
 import { ConversationService } from './service'
 
 export class ConversationApi {
-  constructor(private router: Router, private auth: Auth, private conversations: ConversationService) {}
+  constructor(private conversations: ConversationService) {}
 
-  async setup() {
-    this.router.post(
-      '/conversations',
-      this.auth.client.auth(async (req, res) => {
-        const { error } = Schema.Api.Create.validate(req.body)
-        if (error) {
-          return res.status(400).send(error.message)
-        }
-
-        const { userId } = req.body
-        const conversation = await this.conversations.create(req.client!.id, userId)
-
-        res.send(conversation)
-      })
-    )
+  setup(router: ApiManager) {
+    router.post('/conversations', Schema.Api.Create, this.create.bind(this))
+    router.get('/conversations/:id', Schema.Api.Get, this.get.bind(this))
+    router.get('/conversations/user/:userId', Schema.Api.List, this.list.bind(this))
+    router.get('/conversations/user/:userId/recent', Schema.Api.Recent, this.recent.bind(this))
```

Comment:
The list messages by conversation route was changed from GET /conversations/:userId to GET /conversations/user/:userId. Basically the reasoning behind this is that we are listing all the `/conversations` by their `/user` property with the value `/:userId`. The same reasoning is applied the all other routes

Reply:
The more "object oriented" route `GET /user/:userId/conversations` isn't the name of the route here since it doesn't fit with the way messaging is made. In the messaging codebase services and object have no notion of where they are used. So for example the function to list the conversations of a user is in the ConversationService, not the UserService. It's made in a more bottom-up manner as opposed to top-down. So it makes sense that our routes also follow this logic. `GET /users/:userId/conversations` is the top-down way of doing it while `GET /conversations/user/:userId` is the bottom up way of doing it. @laurentlp let me know what you think

Reply:
Doing `GET /user/:userId/conversations` is a more REST/RESTful way of doing things, but I understand that in our context `GET /conversations/user/:userId` makes more sense.